client/app/scripts/liveblog-edit/module.js

```diff
@@ -135,6 +137,11 @@ define([
                 $scope.panelState = panel;
                 // update url for deeplinking
                 $route.updateParams({panel: $scope.panelState});
+                //clear the new contribution notification
+                if (panel === 'contributions') {
+                    postsService.resetUnreadContributtions(panel);
+                }
+                postsService.setPanelState(panel);
```

Comment:
I agree with that, the number of unread posts depends of  course of what panel you opened. But it's not a good reason to let the post service to be aware of the panels.
Maybe you can put your method and post event listener directly into the controller, or create a new service for unread posts if you think that we will need it from more than one controller.